### ISSUE TITLE
Error when importing "External Data"

### ISSUE THREAD

**Issue by jpylypiw** [Overv/openstreetmap-tile-server#209]
Hi there! First of all thank you for creating and maintaining the docker image.
I currently try to import the german map from geofabrik on a clean Debian 11 System.
The database import works fine, but at the end a script `get-external-data.py` is called which fails connecting to `naciscdn.org` on port 443.

The error logged in the container is:

```
2021-09-01 15:54:04  Analyzing table 'planet_osm_polygon'...
2021-09-01 15:54:05  All postprocessing on table 'planet_osm_polygon' done in 2406s (40m 6s).
2021-09-01 15:54:05  All postprocessing on table 'planet_osm_roads' done in 298s (4m 58s).
2021-09-01 15:54:05  osm2pgsql took 4407s (1h 13m 27s) overall.
+ sudo -u postgres psql -d gis -f indexes.sql
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE INDEX
+ sudo chown -R renderer: /home/renderer/src
+ sudo -u renderer python3 /home/renderer/src/openstreetmap-carto/scripts/get-external-data.py -c /home/renderer/src/openstreetmap-carto/external-data.yml -D /home/renderer/src/openstreetmap-carto/data
INFO:root:Checking table simplified_water_polygons
INFO:root:Checking table water_polygons
INFO:root:Checking table icesheet_polygons
INFO:root:Checking table icesheet_outlines
INFO:root:Checking table ne_110m_admin_0_boundary_lines_land
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 574, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='naciscdn.org', port=443): Max retries exceeded with url: /naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/renderer/src/openstreetmap-carto/scripts/get-external-data.py", line 294, in <module>
    main()
  File "/home/renderer/src/openstreetmap-carto/scripts/get-external-data.py", line 230, in main
    download = s.get(source["url"], headers=headers)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='naciscdn.org', port=443): Max retries exceeded with url: /naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))
+ touch /var/lib/mod_tile/planet-import-complete
+ service postgresql stop
 * Stopping PostgreSQL 12 database server
   ...done.
+ exit 0
```

Does the container work properly after the error or is the import unsuccessful?
Is there a library you can update to get this working or can I just skip the external data?

**Comment by astronautchen** [comment]
> Hi there! First of all thank you for creating and maintaining the docker image.
> I currently try to import the german map from geofabrik on a clean Debian 11 System.
> The database import works fine, but at the end a script `get-external-data.py` is called which fails connecting to `naciscdn.org` on port 443.
> 
> The error logged in the container is:
> 
> ```
> 2021-09-01 15:54:04  Analyzing table 'planet_osm_polygon'...
> 2021-09-01 15:54:05  All postprocessing on table 'planet_osm_polygon' done in 2406s (40m 6s).
> 2021-09-01 15:54:05  All postprocessing on table 'planet_osm_roads' done in 298s (4m 58s).
> 2021-09-01 15:54:05  osm2pgsql took 4407s (1h 13m 27s) overall.
> + sudo -u postgres psql -d gis -f indexes.sql
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> CREATE INDEX
> + sudo chown -R renderer: /home/renderer/src
> + sudo -u renderer python3 /home/renderer/src/openstreetmap-carto/scripts/get-external-data.py -c /home/renderer/src/openstreetmap-carto/external-data.yml -D /home/renderer/src/openstreetmap-carto/data
> INFO:root:Checking table simplified_water_polygons
> INFO:root:Checking table water_polygons
> INFO:root:Checking table icesheet_polygons
> INFO:root:Checking table icesheet_outlines
> INFO:root:Checking table ne_110m_admin_0_boundary_lines_land
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 169, in _new_conn
>     conn = connection.create_connection(
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 96, in create_connection
>     raise err
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 86, in create_connection
>     sock.connect(sa)
> TimeoutError: [Errno 110] Connection timed out
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 699, in urlopen
>     httplib_response = self._make_request(
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 382, in _make_request
>     self._validate_conn(conn)
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
>     conn.connect()
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 353, in connect
>     conn = self._new_conn()
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 181, in _new_conn
>     raise NewConnectionError(
> urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 439, in send
>     resp = conn.urlopen(
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 755, in urlopen
>     retries = retries.increment(
>   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 574, in increment
>     raise MaxRetryError(_pool, url, error or ResponseError(cause))
> urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='naciscdn.org', port=443): Max retries exceeded with url: /naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/renderer/src/openstreetmap-carto/scripts/get-external-data.py", line 294, in <module>
>     main()
>   File "/home/renderer/src/openstreetmap-carto/scripts/get-external-data.py", line 230, in main
>     download = s.get(source["url"], headers=headers)
>   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 555, in get
>     return self.request('GET', url, **kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 542, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 655, in send
>     r = adapter.send(request, **kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 516, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='naciscdn.org', port=443): Max retries exceeded with url: /naturalearth/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7b0b850619d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))
> + touch /var/lib/mod_tile/planet-import-complete
> + service postgresql stop
>  * Stopping PostgreSQL 12 database server
>    ...done.
> + exit 0
> ```
> 
> Does the container work properly after the error or is the import unsuccessful?
> Is there a library you can update to get this working or can I just skip the external data?

metoo!

**Comment by jpylypiw** [comment]
So I invested some time yesterday to try to work around this problem.
The problem is not directly related to this repository, but I want to help the community of the docker image to maybe "fix" the issue.
The root of the problem is, that the content delivery network from "Natural Earth Data" does not deliver any content anymore.
This is investigated in this issue here: https://github.com/nvkelso/natural-earth-vector/issues/574
In this comment https://github.com/nvkelso/natural-earth-vector/issues/574#issuecomment-904038875 there is a Amazon S3 Link which also provides the same file.

So I went and replaced the URL in `external-data.yml` to the following URL and now the import works fine:
https://naturalearth.s3.amazonaws.com/110m_cultural/ne_110m_admin_0_boundary_lines_land.zip
